src/components/cron.ts

```diff
@@ -137,7 +137,8 @@ export const createCoffeeChatCron = (client: Client): CronJob =>
 // Gives Codey coin role to those on the leaderboard list everyday
 export const assignCodeyRoleForLeaderboard = (client: Client): CronJob =>
   new CronJob('0 0 0 */1 * *', async function () {
-    const leaderboard = await getCoinLeaderboard(NUMBER_USERS_TO_ASSIGN_ROLE);
+    // extra 5 members in case there are 5 bots in the leaderboard (as per the production environment)
+    const leaderboard = await getCoinLeaderboard(NUMBER_USERS_TO_ASSIGN_ROLE + 5);
```

Comment:
This might give more users than 10 if there are less than 5 bots that are in the top 10 codey coin users. I would make sure that in the code that is responsible for displaying the leaderboard, you only display the top ten people (right now it displays everyone in the leaderboard).

Reply:
@mcpenguin So the code for getCoinLeaderboard uses a SQL query and I am not sure how I can change the SQL query itself to remove the bots. 
What I can do is something like this - 

```javascript
let users = 0;
while (users < 10) {
  const leaderboard = await getCointLeaderboard(NUMBER_USERS_TO_ASSIGN_ROLE + 5);
 ... // the rest of the code 
    if (memberToUpdate && !memberToUpdate.user?.bot) {
        await updateMemberRole(memberToUpdate, await getRoleName(CODEY_COIN_ROLE_ID), true);
       users++; // add that we have added 1 user
      }
}
```

This can ensure that only 10 users get the role. Let me know your thoughts.

Reply:
I believe we should get the leaderboard before the while loop, but yeah, that was how I was thinking we could proceed.

Reply:
Oh okay cool, yeah I will do that and update the pull request.